File: Creational/Constructor/python/constructor.py
# Constructors are used for instantiating an object.
# It initializes the values to the data members of the class 
# when an object of that class is being created

class Car: 
	# constructor 
	def __init__(self, model, year, miles): 
		# self is a reference to the object itself
		self.model = model
		self.year = year
		self.miles = miles
	
	# method for printing data members 
	def printDetails(self): 
		print(self.model, 'has done', self.miles, 'miles')
  
# instantiating an object of the class
obj = Car("Jeep", 2009, 50000)

# calling the instance method
obj.printDetails() 

File: Creational/Constructor/cpp/constructor.cpp
// Constructors are used for instantiating an object.
// It initializes the values to the data members of the class
// when an object of that class is being created

#include <iostream>
#include <string>

using namespace std;

class Car {
private:
	string model;
	int year;
	int miles;
public:
	// constructor
	Car(string Model, int Year, int Miles) {
		model = Model;
		year = Year;
		miles = Miles;
	}
	// Helper function for printing data members.
	void printDetails() {
		cout<<"Car model: "<<model<<"\n";
		cout<<"Car year: "<<year<<"\n";
		cout<<"Car miles: "<<miles<<"\n";
	}
};

int main() {
	// instantiating an object of the class
	Car myCar("Mercedes", 2010, 50);

	// calling the instance method
	myCar.printDetails();
}

